ibm/resource_ibm_cbr_rule.go

```diff
@@ -0,0 +1,519 @@
+// Copyright IBM Corp. 2021 All Rights Reserved.
+// Licensed under the Mozilla Public License v2.0
+
+package ibm
+
+import (
+	"context"
+	"fmt"
+	"log"
+
+	"github.com/IBM/go-sdk-core/v5/core"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+
+	"github.com/IBM/platform-services-go-sdk/contextbasedrestrictionsv1"
+)
+
+func resourceIBMCbrRule() *schema.Resource {
+	return &schema.Resource{
+		CreateContext: resourceIBMCbrRuleCreate,
+		ReadContext:   resourceIBMCbrRuleRead,
+		UpdateContext: resourceIBMCbrRuleUpdate,
+		DeleteContext: resourceIBMCbrRuleDelete,
+		Importer:      &schema.ResourceImporter{},
+
+		Schema: map[string]*schema.Schema{
+			"description": &schema.Schema{
+				Type:         schema.TypeString,
+				Optional:     true,
+				ValidateFunc: InvokeValidator("ibm_cbr_rule", "description"),
+				Description:  "The description of the rule.",
+			},
+			"contexts": &schema.Schema{
```

Comment:
contexts and resources are required according to API docs..
Please define schema properties according API..

Reply:
```
			"contexts": &schema.Schema{
				Type:        schema.TypeList,
				Optional:    true,
				Description: "The contexts this rule applies to.",
				Elem: &schema.Resource{
					Schema: map[string]*schema.Schema{
						"attributes": &schema.Schema{
							Type:        schema.TypeList,
							Required:    true,
							Description: "The attributes.",
							Elem: &schema.Resource{
								Schema: map[string]*schema.Schema{
									"name": &schema.Schema{
										Type:        schema.TypeString,
										Required:    true,
										Description: "The attribute name.",
									},
									"value": &schema.Schema{
										Type:        schema.TypeString,
										Required:    true,
										Description: "The attribute value.",
									},
								},
							},
						},
					},
				},
			},
```